### PR TITLE
RDoc-2021 - Last Java 5.2 & 5.1 Missing Markdowns

### PR DESCRIPTION
**Pull request by Drorlev**

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+{PANEL:**Static indexes**}
+
+There are a couple of ways to create a `static index` and send it to the server. We can use [maintenance operations](../indexes/creating-and-deploying#using-maintenance-operations) or create a [custom class](../indexes/creating-and-deploying#using-abstractindexcreationtask). You can also [scan an assembly](../indexes/creating-and-deploying#using-assembly-scanner) and deploy all found indexes.
+
+<hr />
+
+### Using AbstractIndexCreationTask
+
+If you are interested in having a **strongly-typed syntax** during index creation, or have an **ability to deploy indexes using assembly scanner**, **avoid hard-coding index names in every query**, then `AbstractIndexCreationTask` should be your choice.
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown:21]: not in java - we can't use strongly typed syntax - every index is created as string.

File: Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Indexes: Creating and Deploying Indexes
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown:1]: this is java - please review and then ping me for review.

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/Indexes/Creating.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                // ...
+                configuration.put("MapTimeoutInSec","30");
+                setConfiguration(configuration);
+            }
+        }
+        //endregion
+    }
+
+
+    private class Users_ByName implements IAbstractIndexCreationTask{
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/Indexes/Creating.java:173]: extend AbstractIndexCreationTask - you don't need all those methods

File: Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown (hunk 1 above)
Comment (Drorlev) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/creating-and-deploying.java.markdown:21]: Done

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/Indexes/Creating.java (hunk 3 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/Indexes/Creating.java:173]: Done

**Review by ml054 — APPROVED** [review]

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown [hunk 4]
```diff
@@ -202,6 +203,42 @@ remote calls.
     {CODE-TAB:java:IncludeCounters counters_region_query_include_multiple_Counters@DocumentExtensions\Counters\Counters.java /}
     {CODE-TABS/}
 
+    ---
+
+###Counters Bulk-Insert  
+`store.BulkInsert` is RavenDB's high-performance data insertion operation.
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:209]: ?

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown [hunk 5]
```diff
@@ -202,6 +203,42 @@ remote calls.
     {CODE-TAB:java:IncludeCounters counters_region_query_include_multiple_Counters@DocumentExtensions\Counters\Counters.java /}
     {CODE-TABS/}
 
+    ---
+
+###Counters Bulk-Insert  
+`store.BulkInsert` is RavenDB's high-performance data insertion operation.  
+Use its `CountersFor` interface's `Increment` method to add or update counters with great speed.
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:210]: ?

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown [hunk 6]
```diff
@@ -202,6 +203,42 @@ remote calls.
...
+* Syntax  
+
+   * `countersFor`
+       {CODE:java CountersFor-definition@DocumentExtensions\Counters\Counters.java /}
+
+        | Parameters | Type | Description |
+        |:-------------|:-------------|:-------------|
+        | `id` | `String` | Document ID |
+
+   *   `Increment`
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:221]: ?

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown [hunk 7]
```diff
@@ -202,6 +203,42 @@ remote calls.
...
+           | Parameters | Type | Description |
+           |:-------------|:-------------|:-------------|
+           | `name` | `String` | Counter Name |
+           | `id` | `String` | Document ID |
+           | `delta` | `long` | Default: 1L |
+
+* Usage Flow  
+
+   * Create a `store.bulkInsert` instance.  
+   * Pass the instance's `CountersFor` interface, the document ID
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:233]: ?

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 8]
```diff
@@ -189,8 +188,30 @@ void test() {
...
+            for (User user : result) {
+                String userId = user.getID();
+
+                CountersBulkInsert countersFor = bulkInsert.countersFor(userId);
+
+                 bulkInsert.countersFor(userId).increment("downloaded", 100);
+            }
+        }
+
+        //endregion
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:211]: cleanup unnecessary empty lines + format code

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 9]
```diff
@@ -341,4 +363,26 @@ public void sample() {
...
+
+    //region Increment-definition
+        public void increment(String id, String name, long delta);
+    //endregion
+    }
 }
+class User {
+    String id;
+
+    public void setName(String marcin) {
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:380]: marcin ?

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown (hunk 4 above)
Comment (Drorlev) [reply Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:209]: Fixed

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown (hunk 5 above)
Comment (Drorlev) [reply Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:210]: Fixed

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown (hunk 6 above)
Comment (Drorlev) [reply Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:221]: Fixed

File: Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown (hunk 7 above)
Comment (Drorlev) [reply Documentation/5.2/Raven.Documentation.Pages/document-extensions/counters/counters-and-other-features.java.markdown:233]: Fixed

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 9 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:380]: Copied from tests,
fixed

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 8 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:211]: Done

File: Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
+import {AbstractCsharpIndexCreationTask, DocumentStore} from "ravendb";
+
+const store = new DocumentStore('http://127.0.0.1:8080', 'Northwind');
+const session = store.openSession();
+
+class CEmployees_ByAttachmentNames {}
```
Comment (ml054) [inline Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js:6]: What's CEmployees? Can we choose more meaningful name?

File: Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+    //endregion
+}
+
+
+
+//region query1
+    //return all employees that have an attachment called "cv.pdf"
+    const employees = session.query(Employees_ByAttachmentNames)
```
Comment (ml054) [inline Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js:69]: missing await?

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 12]
```diff
@@ -189,8 +188,25 @@ void test() {
                 .toList();
             //endregion
         }
+        ArrayList<User> result = new ArrayList<>();
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:191]: List<User> result = new ArrayList<>();

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 13]
```diff
@@ -189,8 +188,25 @@ void test() {
...
+        //region bulk-insert-counters
+        try (IDocumentSession session = docStore.openSession()) {
+            IDocumentQuery<User> query = session.query(User.class)
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:194]: this wont' execute query. Is it what we want? toList is missing?

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 14]
```diff
@@ -189,8 +188,25 @@ void test() {
...
+                    .whereLessThan("age", 30);
+        }
+        
+        try (BulkInsertOperation bulkInsert = docStore.bulkInsert()){
+            for (User user : result) {
+                String userId = user.getID();
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:200]: getId or getID?

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 15]
```diff
@@ -341,4 +358,23 @@ public void sample() {
...
+
+    //region Increment-definition
+        public void increment(String id, String name, long delta);
+    //endregion
+    }
 }
+class User {
+    String id;
+    public void setName(String name) {}
+    public String getID() {
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:375]: getId - >you can generate getters/setters automatically

Comment (ml054) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:375]: https://www.jetbrains.com/help/idea/generating-code.html

File: Documentation/5.2/Samples/nodejs/indexes/creating.js [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+        this.map = "docs.Orders.Select(order => new { " +
+            "    Employee = order.Employee, " +
+            "    Company = order.Company, " +
+            "    Total = Enumerable.Sum(order.Lines, l => ((decimal)((((decimal) l.Quantity) * l.PricePerUnit) * (1M - l.Discount)))) " +
+            "})";
+    }
+
+}
+
+async function main() {
```
Comment (ml054) [inline Documentation/5.2/Samples/nodejs/indexes/creating.js:29]: wrong indent

File: Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js (hunk 10 above)
Comment (Drorlev) [reply Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js:6]: Fixed

File: Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js (hunk 11 above)
Comment (Drorlev) [reply Documentation/5.1/Samples/nodejs/documentExtensions/attachments/indexingAttachments.js:69]: Fixed

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 13 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:194]: Fixed

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 14 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:200]: fixed

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 15 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:375]: Fixed,
Now it is really fast

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java (hunk 12 above)
Comment (Drorlev) [reply Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:191]: Changed to List<User>

**Comment by ml054** [comment]
@Drorlev did you push your changes?

**Comment by Drorlev** [comment]
> 
Know it is all good

File: Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java [hunk 17]
```diff
@@ -335,10 +354,45 @@ public void sample() {
             DatabaseSmugglerExportOptions exportOptions = new DatabaseSmugglerExportOptions();
             //region smuggler_options
             exportOptions.setOperateOnTypes(EnumSet.of(
-                DatabaseItemType.INDEXES,
-                DatabaseItemType.DOCUMENTS,
-                DatabaseItemType.COUNTERS));
+                    DatabaseItemType.INDEXES,
```
Comment (ml054) [inline Documentation/5.2/Samples/java/src/test/java/net/ravendb/DocumentExtensions/Counters/Counters.java:357]: we indent with 4 spaces - please reformat this code with your IDE

**Review by ml054 — APPROVED** [review]

**Pull request merged**